include/ensmallen_bits/spsa/spsa.hpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file spsa.hpp
+ * @author N Rajiv Vaidyanathan
+ * @author Marcus Edel
+ *
+ * SPSA (Simultaneous perturbation stochastic approximation) method for
+ * faster convergence.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_SPSA_SPSA_HPP
+#define ENSMALLEN_SPSA_SPSA_HPP
+
+namespace ens {
+
+/**
+ * Implementation of the SPSA methodT. The SPSA algorithm approximates the
+ * gradient of the function by finite differences along stochastic directions.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Spall1998,
+ *   author  = {Spall, J. C.},
+ *   title   = {An Overview of the Simultaneous Perturbation Method for
+ *              Efficient Optimization},
+ *   journal = {Johns Hopkins APL Technical Digest},
+ *   volume  = {19},
+ *   number  = {4},
+ *   pages   = {482--492},
+ *   year    = {1998}
+ * }
+ * @endcode
+ *
```

Comment:
Remove the extra line here and add something like: https://github.com/mlpack/ensmallen/blob/db6917e1fd304c486ef95398954981f49b071164/include/ensmallen_bits/sgd/sgd.hpp#L54-L56